pkg/bytecode/vm.go

```diff
@@ -20,6 +20,9 @@ var (
 	// produce an invalid result. In Golang, floating point division
 	// by zero produces +Inf, and modulo by zero produces NaN.
 	ErrDivideByZero = fmt.Errorf("%w: division by zero", ErrPanic)
+	// ErrBadRepetition is returned when the right-hand side of the array
+	// repetition operator is not an non-negative integer value.
+	ErrBadRepetition = fmt.Errorf("%w: bad repetition count", ErrPanic)
```

Comment:
this error message is a bit vague, perhaps something like this?
```suggestion
	ErrBadRepetition = fmt.Errorf("%w: array repetition must be a positive integer", ErrPanic)
```

Reply:
i was also thinking of alternatives for the word count here, but couldn't come up with anything. I also like `array repetition must be a non-negative integer`

Reply:
I initially had it more verbose when I had forgotten to check for negative numbers, but simplified it when it got too long. I'll keep this base error the same, but add details where the error is returned.